lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationRequestMapper.java

```
@@ -160,32 +166,60 @@ Applicant createPEPApplicant() {
         AtomicInteger addressId = new AtomicInteger(0);
         personAddresses.forEach(
                 personAddress -> {
-                    Address address = new Address();
+                    Address address = createAddressFromPersonAddress(addressId, personAddress);
+                    addresses.add(address);
+                });
 
-                    address.setId("MAINAPPADDRESS_" + addressId.incrementAndGet());
-                    address.setAddressIdentifier("ADDRESS_" + addressId.get());
+        return addresses;
+    }
 
-                    String addressType = mapAddressType(personAddress.getAddressType());
-                    address.setAddressType(addressType);
+    List<Address> mapPEPCheckSingleCurrentAddress(
+            List<uk.gov.di.ipv.cri.common.library.domain.personidentity.Address> personAddresses) {
+        List<Address> addresses = new ArrayList<>();
+        AtomicInteger addressId = new AtomicInteger(0);
+        personAddresses.forEach(
+                personAddress -> {
+                    if (AddressType.CURRENT == personAddress.getAddressType()) {
+                        Address address = createAddressFromPersonAddress(addressId, personAddress);
+                        addresses.add(address);
+                    }
+                });
 
-                    address.setBuildingNumber(personAddress.getBuildingNumber());
-                    address.setBuildingName(personAddress.getBuildingName());
-                    address.setSubBuilding(personAddress.getSubBuildingName());
-                    address.setStreet(personAddress.getStreetName());
-                    address.setPostTown(personAddress.getAddressLocality());
+        // PEP can only be sent with one address
+        if (addresses.size() > 1) {
+            LOGGER.error("mapPEPCurrentAddress Found {} CURRENT Addresses", addresses.size());
+        }
 
-                    address.setPostal(personAddress.getPostalCode());
+        return addresses;
```

return the first element in the list rather than the entire list

---

For mapSingleAddres, the error message is to ensure that if the requirement is added to the system to support multiple current addresses, that pep does not start silently failing again due to that.
Right now the rest of fraudCRI can support this, but the PEP requests requires a single address only. As this is recent issue there not enough information or direction on which current address we would use in a situation of choosing between current addresses.